### PR TITLE
Allow a callable to be passed to snoop()

### PR DESCRIPTION
**Pull request by zzzeek**
Fixes: #80

### CODE REVIEW

File: pysnooper/pysnooper.py [hunk 1]
```diff
@@ -23,6 +23,9 @@ def write(s):
         def truncate():
             with open(output, 'w') as output_file:
                 pass
+    elif hasattr(output, "__call__"):
```
Comment (alexmojaki) [inline pysnooper/pysnooper.py]:
```suggestion
    elif callable(output):
```

Comment (zzzeek) [reply pysnooper/pysnooper.py]: Yes I wasn't sure if you wanted to use callable() here as it was briefly removed in earlier py3k versions but was later restored

**Comment by cool-RR** [comment]
I like this PR, but I don't like the abstractions you added to the test. Please add your test separately without changing the existing test.

Reading material: https://amir.rachum.com/blog/2017/01/14/explicit-tests/

**Comment by zzzeek** [comment]
thank you for your blog post!

**Comment by cool-RR** [comment]
Great job!

**Pull request merged**